File: python/20260317/fajlkezelo.py
from __future__ import annotations

from pathlib import Path

from kutya import Kutya
from macska import Macska


class FajlKezelo:
    def __init__(self, fajl_utvonal: str) -> None:
        self.fajl_utvonal = Path(fajl_utvonal)

    @staticmethod
    def _igen_nem_to_bool(ertek: str) -> bool:
        return ertek.strip().lower() == "igen"

    def beolvas_allatok(self) -> list:
        allatok = []
        with self.fajl_utvonal.open("r", encoding="utf-8") as fajl:
            for sor in fajl:
                tisztitott_sor = sor.strip()
                if not tisztitott_sor:
                    continue

                adatok = tisztitott_sor.split(";")
                tipus = adatok[0].strip().lower()
                nev = adatok[1].strip()
                eletkor = int(adatok[2].strip())
                bekerulesi_ev = int(adatok[3].strip())
                orokbefogadhato = self._igen_nem_to_bool(adatok[4])

                if tipus == "kutya":
                    allatok.append(Kutya(nev, eletkor, bekerulesi_ev, orokbefogadhato, adatok[5].strip()))
                elif tipus == "macska":
                    benti_macska = self._igen_nem_to_bool(adatok[5])
                    allatok.append(Macska(nev, eletkor, bekerulesi_ev, orokbefogadhato, benti_macska))
                else:
                    raise ValueError(f"Ismeretlen allattipus: {tipus}")

        return allatok

    def orokbefogadhatoak_mentese(self, cel_fajl: str, allatok: list) -> None:
        cel_utvonal = Path(cel_fajl)
        with cel_utvonal.open("w", encoding="utf-8") as fajl:
            for allat in allatok:
                if allat.orokbefogadhato:
                    fajl.write(allat.adatok_szovegesen() + "\n")

File: C#/20260317/Allatmenhely/FajlKezelo.cs
namespace Allatmenhely;

public class FajlKezelo
{
    private readonly string _fajlUtvonal;

    public FajlKezelo(string fajlUtvonal)
    {
        _fajlUtvonal = fajlUtvonal;
    }

    private static bool IgenNemToBool(string ertek)
    {
        return ertek.Trim().Equals("igen", StringComparison.OrdinalIgnoreCase);
    }

    public List<Allat> BeolvasAllatok()
    {
        var allatok = new List<Allat>();

        foreach (var sor in File.ReadLines(_fajlUtvonal))
        {
            if (string.IsNullOrWhiteSpace(sor))
            {
                continue;
            }

            var adatok = sor.Split(';', StringSplitOptions.TrimEntries);
            var tipus = adatok[0].ToLowerInvariant();
            var nev = adatok[1];
            var eletkor = int.Parse(adatok[2]);
            var bekerulesiEv = int.Parse(adatok[3]);
            var orokbefogadhato = IgenNemToBool(adatok[4]);

            if (tipus == "kutya")
            {
                allatok.Add(new Kutya(nev, eletkor, bekerulesiEv, orokbefogadhato, adatok[5]));
            }
            else if (tipus == "macska")
            {
                allatok.Add(new Macska(nev, eletkor, bekerulesiEv, orokbefogadhato, IgenNemToBool(adatok[5])));
            }
            else
            {
                throw new InvalidOperationException($"Ismeretlen allattipus: {tipus}");
            }
        }

        return allatok;
    }

    public void OrokbefogadhatoakMentese(string celFajl, IEnumerable<Allat> allatok)
    {
        var sorok = allatok
            .Where(allat => allat.Orokbefogadhato)
            .Select(allat => allat.AdatokSzovegesen());

        File.WriteAllLines(celFajl, sorok);
    }
}
